sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3741,7 +3741,15 @@ ur_result_t UpdateCommandBufferCommand::enqueueImp() {
     default:
       break;
     }
-    MGraph->updateImpl(Node);
+  }
+
+  // Split list of nodes into nodes per UR command-buffer partition, then
+  // call UR update on each command-buffer partition
+  auto PartitionedNodes = MGraph->getPartitionForNodes(MNodes);
+  auto Device = MQueue->get_device();
+  for (auto It = PartitionedNodes.begin(); It != PartitionedNodes.end(); It++) {
+    auto CommandBuffer = It->first->MCommandBuffers[Device];
+    MGraph->updateKernelsImpl(CommandBuffer, It->second);
```

Comment:
We've gone from a generically named update call to a kernel specific one. How would you envision this looking if we supported more node types than just kernels here?

Reply:
Good question. I've renamed this again to `updateURImpl`, as i guess my thinking as that we'd first find the list of UR updatable nodes for each partition, then for each partition call update on it's command-buffer with those UR updatable nodes. And I've put comments/asserts in for now saying that only kernel nodes are supported for now.